### ISSUE TITLE
[BUG] `optimizer` passed as an instance is not bound to the network

### ISSUE THREAD

**Issue by srupat** [alan-turing-institute/sktime#10990]
**Describe the bug**

`BaseDeepClassifierPytorch` documents `optimizer` as accepting "an instance of an optimizer defined in PyTorch". When an instance is passed, the estimator fits without error but **the network is never updated** — `predict` returns the output of a randomly initialized network.

A PyTorch optimizer captures the concrete parameter tensors it will update in `param_groups` at construction time. `self.network` is only created inside `_fit`, so at `__init__` time there is no network to bind to. `_instantiate_optimizer` then returns the user's object unchanged:

https://github.com/sktime/sktime/blob/b750673cb9c873e1eaabf3adfb8c8f44280ac0ca/sktime/classification/deep_learning/base/_base_torch.py#L615-L617

`_run_epoch` calls `self._optimizer.step()` , which walks `param_groups` and updates whatever tensors the user happened to bind — not `self.network`. Gradients land correctly on `self.network`'s `.grad` fields and are then discarded. There is no way for a caller to construct a correct instance, since the network does not exist until `fit` runs. 

By contrast the `str` branch is correct, because it constructs the optimizer inside `_fit`, after the network exists:
https://github.com/sktime/sktime/blob/b750673cb9c873e1eaabf3adfb8c8f44280ac0ca/sktime/classification/deep_learning/base/_base_torch.py#L605-L606
This affects every estimator deriving from `BaseDeepClassifierPytorch` and`BaseDeepRegressorPytorch`.

**To Reproduce**

```python
import numpy as np, torch
from sktime.classification.deep_learning.mlp._mlp_torch import MLPClassifierTorch

X = np.random.randn(20, 1, 24)
y = np.array(["a", "b"] * 10)

# the network does not exist yet, so the optimizer must be bound to something else
placeholder = torch.nn.Linear(4, 3)
opt = torch.optim.Adam(placeholder.parameters(), lr=0.01)

print("--- optimizer passed as an instance ---")
MLPClassifierTorch(optimizer=opt, num_epochs=5, verbose=True, random_state=0).fit(X, y)

print("--- equivalent optimizer passed as a string ---")
MLPClassifierTorch(optimizer="adam", lr=0.01, num_epochs=5, verbose=True,
                   random_state=0).fit(X, y)
```

Output:

```
--- optimizer passed as an INSTANCE ---
Epoch 1: Loss: 0.6956636488437653
Epoch 2: Loss: 0.6931753516197204
Epoch 3: Loss: 0.6916144222021103
Epoch 4: Loss: 0.6931950002908707
Epoch 5: Loss: 0.6963465094566346
--- equivalent optimizer passed as a STRING ---
Epoch 1: Loss: 1.6750205263495446
Epoch 2: Loss: 1.0345402874052525
Epoch 3: Loss: 1.0676873428747058
Epoch 4: Loss: 3.1677918057199577
Epoch 5: Loss: 0.9844802880656047
```

The instance run sits at `ln(2) = 0.693` — the loss of an untrained binary classifier. 

**Expected behavior**

Either the optimizer actually updates `self.network`, or passing an instance is rejected with a clear error instead of silently producing an unfitted model.

**Comment by srupat** [comment]
@fkiraly @phoeenniixx would appreciate your feedback on this one. There are a few ways to handle this. 
1) Either allow passing optimizers only as str, which is what the schedulers already do
Example:
```python
clf = MLPClassifierTorch(optimizer="adamw", lr=1e-2, num_epochs=5)
```
2) Allow passing a class as parameter and instantiate it during `fit` so that the network is now available
Example:
```python
clf = MLPClassifierTorch(
    optimizer=torch.optim.AdamW,
    lr=1e-2,
    optimizer_kwargs={"weight_decay": 0.05, "betas": (0.9, 0.98)},
    num_epochs=5,
)
```

Which approach do you think is better? I personally feel second is better since it allows for more flexibility (that we are looking for when passing an instance).

**Comment by fkiraly** [comment]
I think we should try to fix the instance case. I see two options:

* bind the parameters to a copy of the instance, e.g., via `add_param_group` (I have not used this method before, but my AI suggested it)
* obtain the parameters from the instance, obtain the class from the instance, and re-initialize together with the network parameters.

Also, please check in the other torch network classes. The logic of coercing optimizers, etc, should happen in only one place - but that is a separate issue. It is, however, possible, that this has already been fixed for other estimator types.

FYI @Faakhir30, have you looked at this?
